src/stateMachines/OID4VCIMachine.tsx

```diff
@@ -0,0 +1,534 @@
+        type: 'final',
+      },
+      [OID4VCIMachineStates.declined]: {
+        id: OID4VCIMachineStates.declined,
+        type: 'final',
+      },
+      [OID4VCIMachineStates.error]: {
+        id: OID4VCIMachineStates.error,
+        type: 'final',
+      },
+      [OID4VCIMachineStates.done]: {
+        id: OID4VCIMachineStates.done,
+        type: 'final',
+      },
+    },
+  });
+};
+
+export class OID4VCIMachine {
+  private static _instance: OID4VCIMachineInterpreter | undefined;
+
+  static hasInstance(): boolean {
+    return !!OID4VCIMachine._instance;
+  }
+
+  static get instance(): OID4VCIMachineInterpreter {
+    if (!this._instance) {
+      throw Error('Please initialize an oid4vci machine first');
+    }
+    return this._instance;
+  }
+
+  static stopInstance(): void {
+    debug(`Stop instance...`);
+    if (!OID4VCIMachine.hasInstance()) {
+      return;
+    }
+    OID4VCIMachine.instance.stop();
+    OID4VCIMachine._instance = undefined;
+    debug(`Stopped instance`);
+  }
+
+  static newInstance(opts?: OID4VCIMachineInstanceOpts): OID4VCIMachineInterpreter {
+    const instance: OID4VCIMachineInterpreter = interpret(
+      createOID4VCIMachine(opts).withConfig({
+        services: {
+          [OID4VCIMachineServices.initiating]: initiating,
+          [OID4VCIMachineServices.createCredentialSelection]: createCredentialSelection,
+          [OID4VCIMachineServices.retrieveContact]: retrieveContact,
+          [OID4VCIMachineServices.retrieveCredentials]: retrieveCredentials,
+          [OID4VCIMachineServices.addContactIdentity]: addContactIdentity,
+          [OID4VCIMachineServices.verifyCredentials]: verifyCredentials,
+          [OID4VCIMachineServices.storeCredentialBranding]: storeCredentialBranding,
+          [OID4VCIMachineServices.storeCredentials]: storeCredentials,
+          ...opts?.services,
+        },
+        guards: {
+          oid4vciHasNotContactGuard,
+          oid4vciSelectCredentialsGuard,
+          oid4vciAuthenticationGuard,
+          oid4vciContactIdentityGuard,
+          oid4vciVerificationCodeGuard,
+          oid4vciHasContactGuard,
+          oid4vciCreateContactGuard,
+          oid4vciHasSelectedCredentialsGuard,
+          ...opts?.guards,
+        },
+      }),
+    );
+
+    if (typeof opts?.subscription === 'function') {
+      instance.onTransition(opts.subscription);
+    } else if (opts?.requireCustomNavigationHook !== true) {
+      instance.onTransition((snapshot: OID4VCIMachineState): void => {
+        void oid4vciStateNavigationListener(instance, snapshot);
+      });
+    }
+
+    return instance;
+  }
+
+  static getInstance(opts?: OID4VCIMachineInstanceOpts & {requireExisting?: boolean}): OID4VCIMachineInterpreter {
+    if (opts?.requireExisting === true) {
+      throw Error(`Existing oid4vci instance requested, but none was created at this point!`);
+    }
+    OID4VCIMachine._instance = OID4VCIMachine.newInstance(opts);
```

Comment:
This always overwrites the instance. If you really want new instances, then the getInstance method is really confusing

Reply:
refactored